### DISCUSSION TITLE
Output file management for batches

### DISCUSSION THREAD

**Discussion by MrFrankDavis** [deforum-art/sd-webui-text2video#231] [Ideas]
I would like to see an update with the same file management system as deforum.  I like to make batch outputs, and it would be nice if all the outputs weren't named vid.

**Comment by terbo** [comment]
Only way I've found to do this is through the API. A pretty simple python script can do this. I'm playing with one right now if you want the code.

**Comment by MrFrankDavis** [comment]
Yes please, that would be awesome!
________________________________

**Reply by terbo** [reply]
[Here](https://github.com/terbo/sdscripts/blob/main/dev/t2v.py) ya go.

Usage:
Enable API access in automatic1111 (add --api to COMMANDLINE_ARGS in webui-user.bat/sh)
In settings/text2video, select "All" under 'Keep model in VRAM between runs'.
Run `python t2v.py -p "your prompt" --api http://127.0.0.1:7860`
Or `python t2v.py -f prompts.txt ...`

Most of the options are available on the command line, however I haven't been able to get img2img/vid2vid to work, as the post requests keep failing.

Have fun!

**Reply by beepbeepplop** [reply]
This is great. But it's not working and giving me tracebacks using the api. Using the extension i can generate videos fine. Any ideas?

**Reply by beepbeepplop** [reply]
Solved it. Solution here: https://github.com/deforum-art/sd-webui-text2video/issues/208

**Reply by beepbeepplop** [reply]
having got it working i've noticed that my vids are really washed out (vae issue?). The same is not true using the extension directly. Getting this working would solve a major thing I am looking for (queueing modelscope with prompts). Any help/advice? Thanks.

**Reply by terbo** [reply]
Did you manually specify a video model and checkpoint? Unfortunately I haven't run into this issue.
I've begun using ComfyUI and think it has a lot more potential, without needing the API or extra scripts.
It's more complex and a lot more _clicking_, but if automatic doesn't cut it, I'd check it out.

If you show me the settings you are using maybe I can help.

**Reply by beepbeepplop** [reply]
Hi Terbo. I have used comfyui too. I am looking to queue prompts for modelscope. That's it really. Can that be done with comfyui? As for specifying video model and checkpoints I ran your script in the commandline. I get a video but it's washed out. What settings do you want to see?

**Reply by terbo** [reply]
Yes, https://github.com/comfyanonymous/ComfyUI/discussions/896 details how, scroll down a bit.
For t2v.py, you may get the best results if you specify as many options as you can, e.g.:

`python ./t2v.py  -W 576 -H 320 --steps 15 --sampler DDIM_Gaussian --fps 8 --frames 24 --cfg 15 -p "darth vader is surfing on waves" --api http://localhost:7861`

This text2video extension hasn't been updated in a few months, and since I wrote this little t2v script
I began using animatediff then switched to comfyui, and the versatility is incomparable.

**Reply by beepbeepplop** [reply]
Still looking washed out to me. https://github.com/deforum-art/sd-webui-text2video/assets/22233124/fc8cda02-c254-41fc-abb5-bdcd0af10f0b
Can you run modelscope in comfyui?

**Reply by terbo** [reply]
That's about the quality that I got at lower resolutions. SVD generates higher quality images.
I'm also working on a flow that takes the last image and tries to continue the animation, which is all I think runwayml does.

Here's what I get:

openjourney/mm v2 ![vadar_00006_](https://github.com/deforum-art/sd-webui-text2video/assets/2475167/a16396ac-ddfe-4c11-b3fa-ffedc5936a0d)

SVD: [svd.mp4](https://github.com/deforum-art/sd-webui-text2video/assets/2475167/c12d80b0-d000-4f35-a8fb-45d8ac7a8500)
The artifacts are from converting it to mp4, it looks normal in webp but haven't figured out how to output to gif yet.

**Reply by beepbeepplop** [reply]
it was more the colours of the clip rather than the quality. the sepia. I guess something vae related has changed since i used to use it.

**Reply by MrFrankDavis** [reply]
I've been noticing a lot of the outputs are washed out too.  I've just made it a habit of color correcting in my video editing program.
https://www.youtube.com/channel/UCoeTLEkXmPzIOLD2lg3DV0A

**Reply by beepbeepplop** [reply]
ah okay. that's good to know. Something has obviously changed because outputs are very washed out compared to what i used to get, but this is defo a solution of sorts...

**Reply by beepbeepplop** [reply]
I've used svd and got some good results. I just think that modelscope is a great model and i've generated lots of great videos that no other model will do.

**Reply by MrFrankDavis** [reply]
I prefer zeroscope to modelscope due to the shutterstock logo.

**Reply by beepbeepplop** [reply]
the shutterstock logo doesn't bother me. zeroscope doesn't know as much. i've been able to generate much "better" videos with modelscope.

**Reply by MrFrankDavis** [reply]
Try adding vivd color to the end of the prompt.  I've seen it make a big difference!

**Reply by beepbeepplop** [reply]
That's not something I've had to do before. I will nuke my install and see if starting from scratch gets me back to working properly...

**Reply by beepbeepplop** [reply]
Tried with another install. Videos are definitely washed out compared to earlier videos. But seems noone is really interested in modelscope anymore?

**Reply by MrFrankDavis** [reply]
Could it have something to do with TensorRT?  I think it started for me near the time I added the extension.

**Reply by beepbeepplop** [reply]
I worked it out. After all my moaning it was was the Sampling method. Set it to DDIM and it works like I expected.

**Comment by MrFrankDavis** [comment]
Thanks!
________________________________
